lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -418,22 +418,21 @@ void ImageDecoderImpeller::UploadTextureToPrivate(
             result(image, decode_error);
           })
           .SetIfTrue([&result, context, buffer, image_info, resize_info] {
-            // The `result` function must be copied in the capture list for each
-            // closure or the stack allocated callback will be cleared by the
-            // time to closure is executed later.
+            auto result_ptr = std::make_shared<ImageResult>(std::move(result));
             context->StoreTaskForGPU(
-                [result, context, buffer, image_info, resize_info]() {
+                [result_ptr, context, buffer, image_info, resize_info]() {
                   sk_sp<DlImage> image;
                   std::string decode_error;
                   std::tie(image, decode_error) =
```

Comment:
Can we remove the double tie while we are at it. Perhaps:

```c++
auto [image, decoder_error] = UploadTex...;
```

Reply:
I'm actually removing any change to this file.  If for some reason I edit this, i'll fix it.

Reply:
Okay, i ended up bringing this back and I removed the extra tie.